docs/running-locally.md

```diff
@@ -1,5 +1,37 @@
 # Running Locally
 
+You have two options:
+
+1. If you're using VSCode, you use the Dev-Container. This takes about 7 minutes. Pre-commit checks take about 4 minutes to run.
+1. Install the requirements on your computer manually. This takes about 30 minutes. Pre-commit checks take about 3 minutes to run.
+
+## Git Clone
+
+Close the Git repo into: `$(GOPATH)/src/github.com/argoproj/argo-workflows`. Any other path will mean the code
+generation does not work.
+
+## Development Container
+
+A development container is a running Docker container with a well-defined tool/runtime stack and its prerequisites.
+[The Visual Studio Code Remote - Containers](https://code.visualstudio.com/docs/remote/containers)  extension lets you use a Docker container as a full-featured development environment.
+
+System requirements can be found [here](https://code.visualstudio.com/docs/remote/containers#_system-requirements)
+
+Note:
+
+* `GOPATH` must be `/go`.
+* for **Apple Silicone**
+  * Configure Docker Desktop to use BuildKit:
+
+    ```json
+    "features": {
+      "buildkit": false
+    },
+    ```
+
+* For **Windows WSL2**
+  * Configure [`.wslconfig`](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#configuration-setting-for-wslconfig) to limit memory usage by the WSL2 to prevent VSCode OOM.
+
```

Comment:
Can you add steps on how to run `make pre-commit -B` in the dev container? I could not figure that out.

Reply:
Do you mean something like and if we need more detail than this?

1.  open powershell and check, make sure your wsl version is 2 `wsl --list --verbose`
2. In wsl, clone argo-workflows to `/home/$USER/go/src/github.com/argoproj/argo-workflows`
3. in vscode, use "Open Folder in Container" to open the step2 location. The Windows explorer path would be something like `\\wsl$\Ubuntu\home\USER\go\src\github.com\argoproj\argo-workflows`
4. you can run `make pre-commit -B` after successful dev-container build and open in vscode